enterprise/app/services/enterprise/billing/handle_stripe_event_service.rb

```diff
@@ -101,10 +124,39 @@ def enable_features_for_current_plan
     enable_plan_specific_features
   end
 
-  def reset_captain_usage
+  def handle_subscription_credits(plan)
+    plan_name = plan['name']
+    plan_credits = get_plan_credits(plan_name)
+
+    # Get current limits
+    current_limits = account.limits || {}
+    current_topup = current_limits['captain_responses_topup'].to_i
+
+    # On subscription renewal/update:
+    # 1. Reset monthly credits to plan allocation
+    # 2. Preserve topup credits (they don't expire)
+    # 3. Reset usage counter
+    monthly_credits = plan_credits[:responses]
+    total_credits = monthly_credits + current_topup
+
+    account.update!(
+      limits: current_limits.merge(
+        'captain_responses_monthly' => monthly_credits,
+        'captain_responses_topup' => current_topup,
+        'captain_responses' => total_credits,
+        'captain_documents' => plan_credits[:documents]
+      )
+    )
+
+    # Reset the usage counter for the new billing period
     account.reset_response_usage
```

Comment:
Will the reset usage also reset the usage of their credits the customers purchased? I think we can remove this call

Reply:
Yeah, that makes credits purchased not `one-time` use. 

Alex buys 5000 credits -> Uses 4000 credits -> Usage resets next month, they have 5000 credits again